### PR TITLE
New PR for factory comparators

### PR DESCRIPTION
**Pull request by AlexChenl3**
due to merging issues with main, PR92 was closed and a new branch was formed from before the merging issues. from PR92:

Creating a factory for comparator classes and switching from ItemComparators to ProductComparators to better adhere to clean architecture. After creating the factory, also tried to clean up intellij errors in Wishlist. This lead me to creating a helper method that is called by the sortProductList methods and fixing the default saved sort order for ReviewCount and ReviewStar. Also added javadocs to some classes that doesnt have them

### CODE REVIEW

**Review by chinchinjim — APPROVED** [review]
Nice job adding documentation + clean arch into the code, makes it a lot more readable now :) some tests aren't passing, but they'll be fixed in a future pr so other than that it looks good

**Review by Ljime — CHANGES_REQUESTED** [review]
The changes still do not comply with clean architecture, we are not allowed to create controllers or anything above the entity layer inside of an entity. The sorting functions should not be in Wishlist. It sthould be in controller where a controller is created and called from the UI

File: src/main/java/Entities/ProductComparatorController.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    /**
+     * Returns a ProductComparator based off word
+     * @param word String value that corresponds to what type of comparator is created
+     * @return ProductComparator that is created in the method depending on word
+     */
+    public ProductComparator sortWishlist (String word) {
+        ProductComparatorFactory ProductFactory = new ProductComparatorFactory();
+
+        switch (word.toLowerCase()) {
+            case "date":
```
Comment (Ljime) [inline src/main/java/Entities/ProductComparatorController.java:16]: This switch case can be simplied to return productFactory.createComparator(word.toLowerCase());

**Review by kevinle623 — DISMISSED** [review]

**Comment by AlexChenl3** [comment]
also TODO: sortListTest and sortingHelperTest in ProductComparatorControllerTest

File: src/main/java/Controller/ProductComparatorController.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+import java.util.Date;
+
+import UseCases.Comparator.ProductComparator;
+import UseCases.Comparator.ProductComparatorFactory;
+
+/**
+ * a controller class that uses comparator factory to create the different types of comparators
+ * and sorts the wishlist using that comparator
+ */
+public class ProductComparatorController implements ProductList {
```
Comment (Ljime) [inline src/main/java/Controller/ProductComparatorController.java:16]: Why does this implement product list and have add and remove item functions? This is a controller just for sorting is it not?

File: src/main/java/Controller/ProductComparatorController.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+                productList.sort(Collections.reverseOrder(productComparator));
+                break;
+            default:
+                throw new IllegalArgumentException();
+        }
+    }
+
+    public String getName() {
+        return this.name;
+    }
```
Comment (Ljime) [inline src/main/java/Controller/ProductComparatorController.java:62]: Why are there getters and setters here?

**Review by Ljime — CHANGES_REQUESTED** [review]

**Review by DavidZ79 — APPROVED** [review]

**Review by Ljime — APPROVED** [review]

**Review by linajin03 — APPROVED** [review]

**Review by PyroPoro — APPROVED** [review]
Good implementation of the factory design pattern! For the tests, it doesnt seem like you need to test two lists that are identical and call the same method on both lists as one is sufficient, but it doesn't hurt to test both.

**Pull request merged**